types/ovenplayer/index.d.ts

```diff
@@ -1,470 +1,469 @@
+        isTimecodeMode(): boolean;
+        getFramerate(): number;
+        seekFrame(frame: number): void;
+        getDuration(): number;
+        getPosition(): number;
+        getVolume(): number;
+        setVolume(volume: number): void;
+        getMute(): boolean;
+        setMute(mute: boolean): void;
+        play(): void;
+        pause(): void;
+        stop(): void;
+        seek(position: number): void;
+        getPlaybackRate(): number;
+        setPlaybackRate(rate: number): void;
+        getPlaylist(): OvenPlayerPlayList;
+        getCurrentPlaylist(): number;
+        setCurrentPlaylist(index: number): void;
+        getSources(): OvenPlayerSource[] | OvenPlayerPlayList;
+        getCurrentSource(): number;
+        setCurrentSource(index: number): void;
+        getQualityLevels(): OvenPlayerQuality[];
+        getCurrentQuality(): number;
+        setCurrentQuality(index: number): void;
+        isAutoQuality(): boolean;
+        setAutoQuality(auto: boolean): void;
+        addCaption(track: Pick<OvenPlayerTrack, "file" | "kind" | "label">): void;
+        getCaptionList(): OvenPlayerTrack[];
+        getCurrentCaption(): string;
+        setCaption?(track: OvenPlayerTrack): void;
+        setCurrentCaption(index: string): void;
+        removeCaption(index: string): void;
+        showControls(show: boolean): void;
+        toggleFullScreen(): void;
+        on: OvenPlayerHandler;
+        once: OvenPlayerHandler;
+        off(eventName: keyof OvenPlayerEvents): void;
+        remove(): void;
+    }
 
-interface OvenPlayerBrowser {
-    browser: string;
-    browserMajorVersion: number;
-    browserVersion: string;
-    cookies: boolean;
-    mobile: boolean;
-    os: string;
-    osVersion: string;
-    screen: string;
-    ua: string;
-}
+    interface OvenPlayerQuality {
+        bitrate: string;
+        height: number;
+        index: number;
+        label: string;
+        width: number;
+    }
 
-interface OvenPlayerTrack {
-    file: string;
-    kind: string;
-    label: string;
-    data: string[];
-    id: string;
-    name: string;
+    interface OvenPlayerBrowser {
+        browser: string;
+        browserMajorVersion: number;
+        browserVersion: string;
+        cookies: boolean;
+        mobile: boolean;
+        os: string;
+        osVersion: string;
+        screen: string;
+        ua: string;
+    }
+
+    interface OvenPlayerTrack {
+        file: string;
+        kind: string;
+        label: string;
+        data: string[];
+        id: string;
+        name: string;
+    }
 }
 
-export {
-    OvenPlayerBrowser,
-    OvenPlayerConfig,
-    OvenPlayerEvents,
-    OvenPlayerHandler,
-    OvenPlayerIceServer,
-    OvenPlayerInstance,
-    OvenPlayerPlayList,
-    OvenPlayerQuality,
-    OvenPlayerSource,
-    OvenPlayerState,
-    OvenPlayerTrack,
-    OvenPlayerWebRTCStream,
-};
+interface OvenPlayerInterface {
+    debug(this: OvenPlayerInterface, debug: boolean): boolean;
```

Comment:
The old interface didn't have `this` params; what required that change?

Reply:
Formerly the interface's members belonged only to the default export and could not be used as a named import.

Using the following as an example: 

`import { create } from 'ovenplayer'`

Resulted in error: 

`Module '"ovenplayer"' has no exported member 'create'.`

--- 

With these changes, those members could be accessed as a named import. Supplying the parameter `this` disallows unbound usage but as a new error:

`The 'this' context of type 'void' is not assignable to method's 'this' of type 'OvenPlayerInterface'.`

This aligns with [OvenPlayer's documentation](https://airensoft.gitbook.io/ovenplayer/initialization) and helps consumers keep context.

Reply:
This would affect existing destructuring after import.

```js
import OvenPlayer from "ovenplayer";

const { create } = OvenPlayer;

create(...) // no longer works
```

If that's considered breaking, I can do a bump or remove.

Reply:
At that point, why not just export those funcs from the namespace and then keep `export = OvenPlayer`?

Reply:
Afraid there could be references to `this` within those methods that should point to `OvenPlayerInterface` and begin breaking. I'm not familiar enough with this library to say that is fine.

This issue seems to suggest it cannot be used that way yet:

https://github.com/AirenSoft/OvenPlayer/issues/322#issuecomment-1888709919

Otherwise seems like the way to go.

Reply:
Does the library actually crash when destructured? I can't seem to find an indication in the source that the object actually needs `this` based on https://unpkg.com/browse/ovenplayer@0.10.37/src/js/ovenplayer.js

Reply:
Nope - I agree with your observation. 

I'll remove the interface change and simplify the exports.

Reply:
0c924509a869195dc5a6004cf8c4d06930567798